src/storage/v2/constraints/constraints.hpp

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#pragma once
+
+#include "storage/v2/config.hpp"
+#include "storage/v2/constraints/existence_constraints.hpp"
+#include "storage/v2/disk/unique_constraints.hpp"
+#include "storage/v2/inmemory/unique_constraints.hpp"
+
+namespace memgraph::storage {
+
+struct Constraints {
+  Constraints(const Config &config, StorageMode storage_mode) {
+    std::invoke([this, config, storage_mode]() {
+      existence_constraints_ = std::make_unique<ExistenceConstraints>();
+      if (storage_mode == StorageMode::IN_MEMORY_TRANSACTIONAL || storage_mode == StorageMode::IN_MEMORY_ANALYTICAL) {
+        unique_constraints_ = std::make_unique<InMemoryUniqueConstraints>();
+      } else {
+        unique_constraints_ = std::make_unique<DiskUniqueConstraints>(config);
```

Comment:
I think using `switch` statement is a better option as in that case when and if we add new storage mode, build will break as we must explicitly cover cases